galsim/wfirst/wfirst_backgrounds.py

```diff
@@ -0,0 +1,175 @@
+for which the main contribution is zodiacal light.
+"""
+
+import galsim
+import numpy as np
+import os
+
+def getSkyLevel(bandpass, position=None, e_lat=None, e_lon=None, exp_time=None):
+    """
+    Get the expected sky level for a WFIRST observation due to zodiacal light for this bandpass and
+    position.
+
+    This routine requires Bandpass objects that were loaded by galsim.wfirst.getBandpasses().  That
+    routine will have stored tables containing the sky background as a function of position on the
+    sky for that bandpass.  This routine then interpolates between the values in those tables to
+    arbitrary positions on the sky.
+
+    The numbers that are stored in the Bandpass object `bandpass` are background level in units of
+    e-/m^2/s/arcsec^2.  To get rid of the m^2, this routine multiplies by the total effective
+    collecting area in m^2, and to get rid of the arcsec^2 it multiplies by the pixel area in
+    arcsec^2.  This will give a result in e-/s/pix.  If the user has passed in an exposure time, then
+    the results are returned in e-/pix, otherwise the results are returned in e-/pix/s and the user
+    must multiply by the exposure time later on themselves.
+
+    The source of the tables that are being interpolated is Chris Hirata's publicly-available WFIRST
+    exposure time calculator (ETC):
+
+        http://www.tapir.caltech.edu/~chirata/web/software/space-etc/
+
+    It nominally returns photons/m^2/s/arcsec^2, but the input bandpasses used internally by the ETC
+    code include the quantum efficiency, to effectively convert to e-/m^2/s/arcsec^2.
+
+    Positions can be specified either with a keyword `position`, which must be a CelestialCoord
+    object, or with both `e_lat` and `e_lon` for the ecliptic coordinates (as Angle instances).
+    Without any of those values supplied, the routine will use a default position that looks
+    sensibly away from the sun.
+
+    @param bandpass     A Bandpass object.
+    @param position     A position, given as a CelestialCoord object.  If None, and `e_lat` and
+                        `e_lon` are not specified, then the routine will use an ecliptic longitude
+                        of 90 degrees (as a fair compromise between 0 and 180), and an ecliptic
+                        latitude of 30 degrees (decently out of the plane of the Earth-sun orbit).
+                        [default: None]
+    @param e_lat        Ecliptic latitude, given as an Angle instance.  Can only be specified if
+                        `position` is not specified, and if specified, then `e_lon` must also be
+                        specified. [default: None]
+    @param e_lon        Ecliptic longitude, given as an Angle instance.  Can only be specified if
+                        `position` is not specified, and if specified, then `e_lat` must also be
+                        specified. [default: None]
```

Comment:
At the time when I was testing these were extremely useful, because I had tabulated the results at specific places in ecliptic latitude and longitude, but I guess most users will not in general be starting with this information.  Removed.